core/message-handling.go

```diff
@@ -112,17 +112,27 @@ type replicaMessageApplier func(msg messages.ReplicaMessage) error
 // concurrently.
 type messageReplier func(msg interface{}) (reply <-chan interface{}, err error)
 
+// generatedUIMessageHandler assigns UI and handles generated message.
+//
+// It assigns and attaches a UI to the supplied message, then takes
+// further steps to handle the message. It is safe to invoke
+// concurrently.
+type generatedUIMessageHandler func(msg messages.MessageWithUI)
+
 // generatedMessageHandler handles generated message.
 //
-// It arranges the supplied message to be delivered to peer replicas,
+// It handles the supplied message, generated by the current replica.
+// The message is assumed to be completely initialized. It is safe to
+// invoke concurrently.
+type generatedMessageHandler func(msg messages.ReplicaMessage)
+
+// generatedMessageConsumer receives generated message.
+//
+// It arranges the supplied message to be delivered to peer replicas
 // or the corresponding client, depending on the message type. The
 // message should be ready to serialize and deliver to the recipients.
-type generatedMessageHandler func(msg interface{})
-
-// generatedUIMessageHandler assigns and attaches a UI to a generated
-// message and arranges it to be delivered to peer replicas. It is
-// safe to invoke concurrently.
-type generatedUIMessageHandler func(msg messages.MessageWithUI)
+// It is safe to invoke concurrently.
+type generatedMessageConsumer func(msg messages.ReplicaMessage)
```

Comment:
I opted to use "consume", as opposed to "apply", because this "consumer" simply arranges messages to be delivered rather than manipulating any piece of the consensus state. So, "consume" here means to get the message and stash it wherever appropriate so that it can be delivered to the appropriate recipients, whenever they are ready to receive it.

Reply:
Moreover, all generated messages are applied by `generatedMessageHandler` to the local replica state. Reusing this term might add some confusion...

Reply:
I understood the meaning and that's fine. Thank you for explanation.